plugins/mcreator-core/blockly/js/extensions_procedure_deps.js

```diff
@@ -0,0 +1,227 @@
+        const fieldValues = {};
+        const fieldValuesDummy = [];
+        while (inputBlock && !inputBlock.isInsertionMarker()) {
+            connections.push(inputBlock.valueConnection_);
+            if (inputBlock.valueConnection_)
+                fieldValues[inputBlock.valueConnection_.sourceBlock_.id] = inputBlock.nameValue_;
+            else
+                fieldValuesDummy.push(inputBlock.nameValue_);
+            inputBlock = inputBlock.nextConnection && inputBlock.nextConnection.targetBlock();
+        }
+        for (let i = 0; i < this.inputCount_; i++) {
+            const connection = this.getInput('arg' + i) && this.getInput('arg' + i).connection.targetConnection;
+            if (connection && connections.indexOf(connection) === -1)
+                connection.disconnect();
+        }
+        this.inputCount_ = connections.length;
+        this.updateShape_();
+        const validators = [];
+        for (let i = 0, j = 0; i < this.inputCount_; i++) {
+            Blockly.Mutator.reconnect(connections[i], this, 'arg' + i);
+            const currentField = this.getField('name' + i);
+            validators.push(currentField.getValidator());
+            currentField.setValidator(null);
+            if (connections[i])
+                currentField.setValue(fieldValues[connections[i].sourceBlock_.id] || 'dependency' + i);
+            else if (j < fieldValuesDummy.length)
+                currentField.setValue(fieldValuesDummy[j++] || 'dependency' + i);
+            else
+                currentField.setValue('dependency' + i);
+        }
+        const validNames = [];
+        for (let i = 0, j = 1; i < this.inputCount_; i++) {
+            const currentField = this.getField('name' + i);
+            let currentValue = currentField.getValue();
+            while (validNames.indexOf(currentValue) !== -1)
+                currentValue = 'dependency' + (j++);
+            validNames.push(currentValue);
+            currentField.setValue(currentValue);
+            currentField.setValidator(validators[i]);
+        }
+    },
+
+    saveConnections: function (containerBlock) {
+        let inputBlock = containerBlock.getInputTargetBlock('STACK');
+        let i = 0;
+        while (inputBlock) {
+            if (!inputBlock.isInsertionMarker()) {
+                const input = this.getInput('arg' + i);
+                if (input) {
+                    inputBlock.valueConnection_ = input.connection.targetConnection;
+                    inputBlock.nameValue_ = this.getFieldValue('name' + i);
+                }
+                i++;
+            }
+            inputBlock = inputBlock.getNextBlock();
+        }
+    },
+
+    updateShape_: function () {
+        for (let i = 0; i < this.inputCount_; i++) {
+            if (!this.getInput('arg' + i)) {
+                const validator = uniqueValueValidator('name');
+                const nameField = validOnLoad(new FieldJavaName('dependency' + i, validator));

```

Comment:
If I remove the call to validOnLoad here, nothing really changes. If possible, please provide steps on how can I create a situation where problems happen if this function is not called.

Reply:
So with recent changes this function has become useless?

Reply:
For instance, if there is a block with `dependency1` and `dependency0` defined (the order matters) of different types, opening procedure with such block _without_ this function would swap dependency names. I only did a quick test without real procedure specified (`No entry selected`), but this can also disconnect dependency values from the block and errors would pop up despite the procedure was saved without any.

Reply:
Could you attach sample ptpl?

Reply:
You'll have to change file extension: [:arrow_down:](https://github.com/MCreator/MCreator/files/15118090/ValidOnLoad.ptpl.txt)

Reply:
![image](https://github.com/MCreator/MCreator/assets/16374228/a1257e07-acdc-4c72-b67c-60753390ad35)

I have opened procedure several times and names do not appear to swap. Am I missing something?

Reply:
I have also observed you also remove validators here while assembling the block:

![image](https://github.com/MCreator/MCreator/assets/16374228/48901dec-fa3e-4f66-b9d5-0341f52e84e5)

Isn't this a bit redundant?

----------

Bottom line is, the whole thing with validators seems a bit sus to me

Reply:
There is one. DL selectors.
Only I'm not sure yet how to make them avoid suggesting dependencies already defined. Store them in an array on the block? Just regular selector fields and disable the block if duplicate dependencies are found?

Reply:
So you can't prevent reordering of mutator field order when loading saved?

Because DL selectors sound like overkill compared to "validOnLoad"

I am not asking this because I want to make things more complicated (e.g. DL selectors), but more simple (no need for "validOnLoad" and additionally all code in screenshot above that appears to do similar thing anyway)

Reply:
:heavy_check_mark: Fixed